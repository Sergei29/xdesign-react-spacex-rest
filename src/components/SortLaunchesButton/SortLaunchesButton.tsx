import React from "react";

const SortLaunchesButton = () => {
  return <div></div>;
};

export default SortLaunchesButton;
