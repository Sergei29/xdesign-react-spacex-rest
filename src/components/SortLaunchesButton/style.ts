import styled from "styled-components";

export const SortButtonContainer = styled.button`
  width: 144px;
  height: 43px;
  display: flex;
  justify-content: center;
  align-items: center;
  column-gap: 8px;
  background-color: ${(props) => props.theme.palette.primaryBlue};
  color: ${(props) => props.theme.font.light};
  border: none;
  font-size: 14px;
  font-weight: 400;
  font-family: ${(props) => props.theme.fontFamily};
  &:disabled {
    opacity: 0.7;
  }
  &:active,
  &:focus {
    outline: none;
  }
`;
