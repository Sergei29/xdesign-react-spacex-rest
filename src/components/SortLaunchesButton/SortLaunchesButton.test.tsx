import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import SortLaunchesButton from "./SortLaunchesButton";

describe("SortLaunchesButton", () => {
  let wrapper: ShallowWrapper;
  beforeEach(() => {
    const objMockProps = {
      bDescending: true,
      bDisabled: false,
      funcHandleClick: jest.fn(),
    };
    wrapper = shallow(<SortLaunchesButton {...objMockProps} />);
  });

  it("should render", () => {
    expect(wrapper.length).toEqual(1);
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
