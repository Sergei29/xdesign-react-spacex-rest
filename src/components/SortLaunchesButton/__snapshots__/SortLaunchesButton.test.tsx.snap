// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortLaunchesButton should match snapshot 1`] = `ShallowWrapper {}`;
